backend/chat/models.py

```diff
@@ -0,0 +1,91 @@
+import os
+
+from config.settings import CHOICE_ROOM
+from django.contrib.auth import get_user_model
+from django.db import models
+from PIL import Image
+
+
+class Chat(models.Model):
+    """Chat table"""
+
+    title = models.CharField(max_length=70)
+    room = models.CharField(choices=CHOICE_ROOM, max_length=50)
+    img = models.ImageField(
+        null=True,
+        blank=True,
+        default="default-user-avatar.jpg",
+        upload_to="chat_img",
+    )
+    creator = models.ForeignKey(get_user_model(), null=True, on_delete=models.CASCADE)
+    description = models.TextField(blank=True, null=True)
+    url = models.CharField(max_length=255, blank=True, null=True)
+    timestamp = models.DateTimeField(auto_now_add=True)
+
+    def __str__(self):
+        return f"id: {self.pk}, title: {self.title}, room: {self.room}"
+
+    def get_url(self):
+        """Return absolute url of chat"""
+
+        return f"/chat/{self.room}/{self.pk}/"
+
+    def save(self, *args, **kwargs):
+        """Redefined the save method to save the absolute URL in the url field"""
+
+        # Save default chat img
+        if not self.img:
+            self.img = "default-user-avatar.jpg"
+
+        # Save chat to DB
+        super().save(*args, **kwargs)
+        # Save url
+        self.url = self.get_url()
```

Comment:
The chat was saved to DB before setting the URL.
Probably the URL will not  be saved now.